add more dump_rate info to docs (#500)
POSYDON-code/POSYDON: docs/_source/components-overview/pop_syn/population_params.rst

  * - ``dump_rate``
    - | Batch save after evolving N binaries.

>     - | To facilitate I/O performance, this should not be set below 500 for populations of 100.000 binaries or more. 

Reviewer: Maybe change it to a positive phrasing?

```suggestion
    - | To facilitate I/O performance, this should be higher than 500 for populations of 100.000 binaries or more.
```
